File: benchmarks/cross-language-v1/tasks/concurrent-delta-merge-v1/public/typescript/candidate.ts
export function mergeConcurrentDeltas(base: number, deltaA: number, deltaB: number): number {
  const total = base + deltaA + deltaB;
  if (total < 0) return 0;
  if (total > 1_000_000) return 1_000_000;
  return total;
}

File: benchmarks/cross-language-v1/tasks/concurrent-delta-merge-v1/public/rust/candidate.rs
pub fn merge_concurrent_deltas(base: i64, delta_a: i64, delta_b: i64) -> i64 {
    let total = base + delta_a + delta_b;
    if total < 0 {
        0
    } else if total > 1_000_000 {
        1_000_000
    } else {
        total
    }
}
